### PR TITLE
Upgrade selfhosted compilation to beta.2, and fix transform bug.

### PR DESCRIPTION
**Pull request by loganfsmyth**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues             | 
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  | N
| Minor: New Feature?      | 
| Tests Added/Pass?        | 
| Spec Compliancy?         | 
| License                  | MIT
| Doc PR                   | 
| Any Dependency Changes?  | 

1. Fixed a bug in the `typeof-symbol` transform that I introduced in https://github.com/babel/babel/pull/5706/files#diff-08e96049279b483e87c80d84a7de5166R36
2. Upgrade Babel to selfhost with the most recent beta, with a small workaround for the previously-mentioned bug.

### CODE REVIEW

File: packages/babel-plugin-transform-es2015-typeof-symbol/src/index.js [hunk 1]
```diff
@@ -33,7 +33,9 @@ export default function({ types: t }) {
         const isUnderHelper = path.findParent(path => {
           return (
             (path.isVariableDeclarator() && path.node.id === helper) ||
-            (path.isFunctionDeclaration() && path.node.id.name === helper.name)
```
Comment (jridgewell) [inline packages/babel-plugin-transform-es2015-typeof-symbol/src/index.js]: We need this for `instanceof`, too.

Comment (loganfsmyth) [reply packages/babel-plugin-transform-es2015-typeof-symbol/src/index.js]: good catch, thanks!

**Comment by loganfsmyth** [comment]
Okay this is blocked on https://github.com/istanbuljs/istanbuljs/issues/92

**Comment by loganfsmyth** [comment]
Gonna close until we get the coverage bug fixed.

**Pull request closed without merging**